### PR TITLE
modules: mcux: fix HAS_CMSIS_CORE selection

### PR DESCRIPTION
**Pull request by iuliana-prodan**
Do not select HAS_CMSIS_CORE for SOC_FAMILY_NXP_ADSP. This soc family refers to the Audio DSP from i.MX MPU, which is a HiFi4 core, Xtensa arch.

HAS_CMSIS_CORE is for ARM cores (A, M, R cores) -
see modules/cmsis/Kconfig.

### CODE REVIEW

**Comment by jhedberg** [comment]
@iuliana-prodan could you fix the compliance check failure? Thanks.

File: modules/Kconfig.mcux [hunk 1]
```diff
@@ -5,7 +5,8 @@
 
 config HAS_MCUX
 	bool
-	select HAS_CMSIS_CORE
+	select HAS_CMSIS_CORE if SOC_FAMILY_KINETIS || SOC_FAMILY_IMX || \
```
Comment (manuargue) [inline modules/Kconfig.mcux:8]: considering all other families except ADSP needs this symbol, wouldn't be simpler to do `select HAS_CMSIS_CORE if !SOC_FAMILY_NXP_ADSP`

**Comment by iuliana-prodan** [comment]
> @iuliana-prodan could you fix the compliance check failure? Thanks.

done

**Review by manuargue — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Comment by marc-hb** [comment]
Thanks for the super quick fix! Next time please mention
8d5ed530d7131 in the commit message :-)

**Comment by marc-hb** [comment]
I'm not a Kconfig expert but you have `!SOC_FAMILY_NXP_ADSP` on one line and then `SOC_FAMILY_NXP_ADSP` is still on the next line...?

Also, I wonder whether the `SOC_FAMILY_IMX` makes sense.

**Comment by iuliana-prodan** [comment]
> I'm not a Kconfig expert but you have `!SOC_FAMILY_NXP_ADSP` on one line and then `SOC_FAMILY_NXP_ADSP` is still on the next line...?

No expert either, but here's my explanation for adding `SOC_FAMILY_NXP_ADSP`
-  `!SOC_FAMILY_NXP_ADSP` I've added it to not select the CMSIS_CORE for ADSP since this refers to Xtensa.
-  `SOC_FAMILY_NXP_ADSP` on next line is used in order to be able to select HAS_MCUX when having a board from `SOC_FAMILY_NXP_ADSP` (or simply include mcux-sdk from nxp_hal when compiling nxp_adsp_* boards - part of `SOC_FAMILY_NXP_ADSP`)

Not sure if this PR was the best option - another fix was proposed here: https://github.com/zephyrproject-rtos/zephyr/pull/63558

> Also, I wonder whether the `SOC_FAMILY_IMX` makes sense.

We shouldn't select `CMSIS_CORE` for `SOC_FAMILY_IMX`?
If I understand correctly `SOC_FAMILY_IMX` refers to ARM64 cores and `CMSIS_CORE` is only for ARM32 cores.
If so, yes, `CMSIS_CORE` shouldn't be selected for `SOC_FAMILY_IMX`. 
Please correct me if I'm wrong.

The fix proposed above includes also the second part - do not select `CMSIS_CORE` for `SOC_FAMILY_IMX`.

**Pull request merged**